src/components/dashboard/ActionsList.js

```diff
@@ -0,0 +1,196 @@
+            }
+         }
+
+         .main-row-title {
+            a {
+               font-size: 14px;
+            }
+         }
+      }
+   }
+`
+
+const ActionsList = ({ transaction, actions, wide }) => 
+   actions
+      .slice(0,5)
+      .map((a, i) => (
+         <ActionRow 
+            key={`action-${i}`} 
+            transaction={transaction} 
+            action={a} 
+            actionKind={Object.keys(a)[0]}  
+            wide={wide}
+            i={i}
+         />
+))
+
+const ActionRow = ({ transaction, action, actionKind, wide, showSub = false, toggleShowSub, showSubOpen, i }) => (
+   <CustomGridRow
+      verticalAlign='middle'
+      className={`${wide ? `wide` : ``} ${
+         showSub && showSubOpen === i ? `dropdown-down` : ``
+      } ${showSub ? `showsub` : ``}`}
+      onClick={() => wide && toggleShowSub(i, action)}
+   >
+      <Grid.Column
+         computer={wide ? 15 : 16}
+         tablet={wide ? 14 : 16}
+         mobile={wide ? 14 : 16}
+      >
+         <Grid verticalAlign='middle'>
+            <Grid.Column className='col-image'>
+               <ActionIcon actionKind={actionKind} />
+            </Grid.Column>
+            <Grid.Column className='main-row-title color-black border-bottom'>
+               <ActionMessage 
+                  transaction={transaction}
+                  action={action}
+                  actionKind={actionKind}
+               />
+            </Grid.Column>
+         </Grid>
+      </Grid.Column>
+      {wide && (
+         <Grid.Column
+            computer={1}
+            tablet={2}
+            mobile={2}
+            textAlign='right'
+         >
+            <Image
+               src={showSub && showSubOpen === i ? ArrowBlkImage : ArrowRight}
+               className='dropdown-image dropdown-image-right'
+            />
+            {/* <span className='font-small'>{row[3]}</span> */}
+         </Grid.Column>
+      )}
+   </CustomGridRow>
+)
+
+const ActionMessage = ({ transaction, action, actionKind }) => (
+   <Fragment>
+      {actionKind === 'createAccount' && `New account created: ${transaction.receiverId}`}
+      {actionKind === 'deleteAccount' && `Account deleted: ${transaction.receiverId}`}
+      {actionKind === 'deployContract' && `Contract deployed: ${transaction.receiverId}`}
+      {actionKind === 'functionCall' && `Called method in contract: ${action.functionCall.methodName}`}
+      {actionKind === 'transfer' && `Transferred ${action.transfer.deposit}Ⓝ to ${transaction.receiverId}`}
+      {actionKind === 'stake' && `Staked ${action.stake.stake}Ⓝ`}
+      {actionKind === 'addKey' && (
+         typeof action.accessKey.permission === 'object'
+            ? `Access key added for contract: ${action.accessKey.permission.functionCall.receiverId}`
+            : `New key added for ${transaction.receiverId}`
+      )}
+      {actionKind === 'deleteKey' && `Key deleted`}
+   </Fragment>
+)
+
+const ActionIcon = ({ actionKind }) => (
+   <div>
```

Comment:
any reason to do this vs something more generic?

For example:

```jsx
<img src={`images/action-${actionKind}`} />
```

is there long-term plan to pass stuff like icon color as parameter (so SVG must be inlined)?

Reply:
Yes, Jake proposed to make all icons in one color and color change option ultimately.